sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosPagedIterableTest.java

```diff
@@ -113,6 +117,121 @@ public void readAllItemsBySubscribeWithCosmosPagedIterableHandler() throws Excep
         assertThat(handleCount.get() >= 1).isTrue();
     }
 
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, enabled = false)
+    public void queryItemsWithCosmosPagedIterable() throws Exception {
+
+        CosmosQueryRequestOptions cosmosQueryRequestOptions = new CosmosQueryRequestOptions();
+        cosmosQueryRequestOptions.setMaxBufferedItemCount(10);
+        CosmosPagedIterable<ObjectNode> cosmosPagedIterable = cosmosContainer.queryItems("select * from c",
+                cosmosQueryRequestOptions, ObjectNode.class);
+
+        Iterable<FeedResponse<ObjectNode>> feedResponses = cosmosPagedIterable.iterableByPage(10);
+        //  Just creating iterator drains all the results!
+        Iterator<FeedResponse<ObjectNode>> iterator = feedResponses.iterator();
+        if (iterator.hasNext()) {
+            FeedResponse<ObjectNode> next = iterator.next();
+            logger.info("Next is : {}", next.getResults().size());
+        }
+        Thread.sleep(5 * 1000);
+    }
+
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, enabled = false)
+    public void queryItemsWithCosmosPagedFlux() throws Exception {
+
+        CosmosQueryRequestOptions cosmosQueryRequestOptions = new CosmosQueryRequestOptions();
+        cosmosQueryRequestOptions.setMaxBufferedItemCount(10);
+        CosmosAsyncContainer cosmosAsyncContainer = CosmosBridgeInternal.getCosmosAsyncContainer(cosmosContainer);
+        CosmosPagedFlux<ObjectNode> cosmosPagedFlux = cosmosAsyncContainer.queryItems("select * from c",
+                cosmosQueryRequestOptions, ObjectNode.class);
+
+        CosmosPagedIterable<ObjectNode> cosmosPagedIterable = new CosmosPagedIterable<>(cosmosPagedFlux, 10, 1);
+        Iterator<FeedResponse<ObjectNode>> iterator = cosmosPagedIterable.iterableByPage().iterator();
+        if (iterator.hasNext()) {
+            FeedResponse<ObjectNode> next = iterator.next();
+            logger.info("Next is : {}", next.getResults().size());
+        }
+        Thread.sleep(5 * 1000);
+    }
+
+    @Test(groups = {"unit"})
+    public void validatePrefetchControl() {
```

Comment:
non-blocking -  but I wonder whether we can add a test to validate how many requests we have sent to backend. 
The test is definitely good, but it will not catch any future changes in our pipelines, like adding a flatmap somewhere etc. What if we utilize the metrics, we might have insights how many requests we have sent to backend